### PR TITLE
fix: Panic on nodeClassRef `nil` dereference 

### PR DESCRIPTION
**Pull request by engedaam**
Fixes https://github.com/aws/karpenter-provider-aws/issues/6867

**Description**
- As part of server side apply, we saw that the API Server will send particle objects to be converted by the conversion webhooks. The API server will send a particle objects of fields that are owned by the field manger in applying a patch https://kubernetes.io/docs/reference/using-api/server-side-apply/#field-management. This can cause out conversion webhooks trying to deference a nil point on the nodeClassRef.

**How was this change tested?**
- make presubmit

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
Are we going to add testing for this case?

File: pkg/apis/v1/nodeclaim_conversion.go [hunk 1]
```diff
@@ -63,13 +64,16 @@ func (in *NodeClaimSpec) convertTo(v1beta1nc *v1beta1.NodeClaimSpec, kubeletAnno
 	})
 	// Convert the NodeClassReference depending on whether the annotation exists
 	v1beta1nc.NodeClassRef = &v1beta1.NodeClassReference{}
-	if nodeClassReferenceAnnotation != "" {
-		if err := json.Unmarshal([]byte(nodeClassReferenceAnnotation), v1beta1nc.NodeClassRef); err != nil {
-			return fmt.Errorf("unmarshaling nodeClassRef annotation, %w", err)
+	if v1beta1nc.NodeClassRef != nil {
```
[the remark this answers is not in the document]
Comment (engedaam) [reply pkg/apis/v1/nodeclaim_conversion.go]: That needs to check the version v1 version. Updated the check to catch that case

**Comment by engedaam** [comment]
I am planning on testing the panic of SSA as part E2E testing. But for now I added tests that validate the `nil` check until then.

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

**Comment by anzinchenko89** [comment]
@engedaam, could you please clarify when this fix is supposed to be released and which patch version v1.0.x is going to contain it? Thanks

**Comment by engedaam** [comment]
We don't have a release date, but should be soon. It will be included in the next patch release

**Comment by archana-punia** [comment]
Hi, Is this released in 1.0.4?

**Pull request merged**